django/book/views.py

```diff
@@ -2,10 +2,29 @@
 from django.http import JsonResponse
 from .models import Book
 import json
+from django.views.generic import ListView, DetailView, CreateView
+from django.urls import reverse_lazy
 
 
-# Create your views here.
-def book_list(request):
-    books = Book.objects.all()
-    book_list = list(books.values())
-    return JsonResponse(book_list, safe=False)
+class BookListView(ListView):
+    model = Book
+    template_name = 'book_list.html'
+
+
+class BookDetailView(DetailView):
+    model = Book
+    template_name = 'book_detail.html'
+
+
+class BookCreateView(CreateView):
+    model = Book
+    template_name = 'book_form.html'
+    fields = '__all__'
+    success_url = reverse_lazy('book_list')
+
+
+# # Create your views here.
```

Comment:
I think this commented code could be removed.
Aaah, probably, it has been left here as an example of function-based views